.github/workflows/daily-workflow-ci.yml

```diff
@@ -0,0 +1,54 @@
+name: Daily Workflow CI
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'database/scripts/lib/*'
+      - 'database/scripts/generate_report.rb'
+      - 'database/scripts/format_report.rb'
+jobs:
+  setup:
+    name: Setup
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Ruby 3.2
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+      - name: Extract branch name
+        run: echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+        id: extract_branch
+  generate-report:
+    needs: setup
+    name: Generate Report
+    runs-on: ubuntu-22.04
```

Comment:
Was it a deliberate decision to divide this workflow in three parts requiring to specify in each the type of runner / name / dependencies from each steps?

IMO, it would be less code if you just state a workflow that does all the tasks, given that the amount of those is small, dependent of previous steps and there are no conditional steps. 

I would like to get Clara's input here jic before merging given that I'm not a seasoned infra folk (yet).

Reply:
I'm not against the multiple jobs, but I do think we should try and unify generate-report and format-report. Both push to artifacts and it's unclear to me why we would need a different action. 
As for the setup I'm fine keeping it separate to have a logic separation here on what is setup vs actual execution.

Reply:
I put everything in side the same job so we use the same runner for all tasks: f6bcf6f